File: easy/One-Zero-Two-Zeros.../One Zero, Two Zeros.cs
using System.IO;
using System.Collections.Generic;

class Program
{
    static void Main(string[] args)
    {
        using (StreamReader reader = File.OpenText(args[0]))
        while (!reader.EndOfStream)
        {
            string line = reader.ReadLine();
            if (null == line)
                continue;
            // do something with line
            ShowResult(line);
        }
    }
    
    static void ShowResult(string line){
        int pos = line.IndexOf(' ');
        int range = System.Convert.ToInt32(line.Substring(0,pos));
        int top = System.Convert.ToInt32(line.Substring(pos+1));
        int result = 0;
        for(int i=1;i<=top;i++){
            if (System.Convert.ToString(i, 2).Split('0').Length-1 == range) result++;
        }
        System.Console.WriteLine(result);
    }
}

File: easy/One-Zero-Two-Zeros.../One Zero, Two Zeros.java
import java.io.*;
public class Main {
    public static void main (String[] args) throws IOException {
        File file = new File(args[0]);
        BufferedReader buffer = new BufferedReader(new FileReader(file));
        String line;
        while ((line = buffer.readLine()) != null) {
            line = line.trim();
            // Process line of input Here
            ShowResult(line);
        }
    }
    
    static void ShowResult(String line){
        int pos = line.indexOf(' ');
        int range = Integer.valueOf(line.substring(0,pos));
        int top = Integer.valueOf(line.substring(pos+1));
        int result = 0;
        for(int i=1;i<=top;i++){
            line  = Integer.toBinaryString(i);
            if (line.length() - line.replace("0", "").length() == range) result++;
        }
        System.out.println(result);
    }
}
